src/entities/responsemessage.ts

```diff
@@ -1,7 +1,25 @@
+import { Utils } from "./../utils";
+
 /** @hidden */
 export class ResponseMessage {
-    public result: string; // Result
+    public result?: string; // Result
     public exceptionType?: string; // ExceptionType
+
+    public constructor(event: unknown) {
+        if (typeof event === "object" && event !== null) {
+            if (Utils.hasStringProperty(event, "result")) {
+                this.result = event.result;
+            }
+            if (Utils.hasStringProperty(event, "exceptionType")) {

```

Comment:
I guess that depends on how you define type-safe. In this case, `Utisl.hasStringProperty` actually makes heavy use of the TypeScript type system to propagate type information. 

But I see what you mean, this uses quite a lot of reflection. Let me think about this.

Reply:
I'll leave this as-is for now, since `Utils.hasStringProperty`  already makes heavy use of TypeScript's type safety features, as I mentioned. 

Unfortunately, since the input to `ResponseMessage`'s constructor is a task's result, there's no easy way to constraint that datatype. The only alternative would be to move these property checks elsewhere, but that'd be just moving the reflection checks around :)